compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -175,6 +176,9 @@ public Symbol getBCompiledSymbol(BSymbol symbol, String name) {
             if (symbol.owner instanceof BObjectTypeSymbol) {
                 return createObjectFieldSymbol((BVarSymbol) symbol);
             }
+            if (symbol.owner instanceof BTypeSymbol && Symbols.isFlagOn(symbol.flags, Flags.FIELD)) {
```

Comment:
`symbol.owner` can be extracted out to a variable, right? Not introduced by this PR, but why aren't we using the symbol kind where possible, instead of `instanceof` checks?

Reply:
We were hoping to do a revamp later for this. We will do these changes with a separate PR (Revamp SymbolFactory/TypeFactory)

Reply:
Ack.